Annex60/Utilities/Time/CalendarTime.mo

```diff
@@ -0,0 +1,279 @@
+  else
+    timOff :=0;
+    // this code should not be reachable
+    assert(false, "No valid TimeReference could be identified.
+   This is a bug, please submit a bug report for the this model.");
+  end if;
+
+  // add additional offset when using a custom date and time
+  if timRef == Annex60.Utilities.Time.BaseClasses.TimeReference.Custom then
+    timOff :=timOff + ((dayRef - 1) + sum({dayInMonth[i] for i in 1:(monthRef - 1)})
+     + (if monthRef > 2 and isLeapYear[yearRef - firstYear + 1] then 1 else 0))*3600*24 +
+    3600*hourRef + 60*minuteRef + secondRef;
+  end if;
+
+   // input data range checks at initial time
+  assert(tim + timOff >= timeStampsNewYear[1],
+    if timRef == Annex60.Utilities.Time.BaseClasses.TimeReference.UnixTimeStamp then
+      "Could initialise date in the CalendarTime block.
+   You selected 1970 as the time=0 reference.
+   Therefore the simulation startTime must be at least " + String(timeStampsNewYear[1]) + "."
+    elseif timRef == Annex60.Utilities.Time.BaseClasses.TimeReference.Custom then
+      if yearRef <firstYear then
+        "Could not initialise date in the CalendarTime block.
+   You selected a custom time=0 reference.
+   The minimum value for yearRef is then " + String(firstYear) + " but your value is " + String(yearRef) + "."
+      else
+        "Could not initialise date in the CalendarTime block.
+   You selected a custom time=0 reference.
+   Possibly your startTime is too small."
+      else
+        "Could not initialise date in the CalendarTime block.
+   Possibly your startTime is negative?");
+
+  assert(tim + timOff < timeStampsNewYear[size(timeStampsNewYear,1)],
+    if timRef == Annex60.Utilities.Time.BaseClasses.TimeReference.Custom and yearRef >= lastYear then
+      "Could not initialise date in the CalendarTime block.
+   You selected a custom time=0 reference.
+   The maximum value for yearRef is then " + String(lastYear) + " but your value is " + String(yearRef) + "."
+    else
+       "Could not initialise date in the CalendarTime block.
+       Possibly your startTime is too large.");
+
+  // iterate to find the year at initialisation
+  year :=0;
+  for i in 1:size(timeStampsNewYear,1) loop
+    if unixTimeStamp < timeStampsNewYear[i] then
+      yearIndex :=i - 1;
+      year :=firstYear + i - 2;
+      break;
+    end if;
+  end for;
+
+  // iterate to find the month at initialisation
+  epochLastMonth := timeStampsNewYear[yearIndex];
+  for i in 1:12 loop
+    if (unixTimeStamp-epochLastMonth)/3600/24 <
+      (if i==2 and isLeapYear[yearIndex] then 1 + dayInMonth[i] else dayInMonth[i]) then
+      month :=i;
+      break;
+    else
+      epochLastMonth :=epochLastMonth + (if i == 2 and isLeapYear[yearIndex]
+         then 1 + dayInMonth[i] else dayInMonth[i])*3600*24;
+    end if;
+  end for;
+
+equation
+  // compute unix time step based on found offset
+  unixTimeStamp = tim + timOff;
+
+  // fixme: Both "when" constructs need to be rewritten.
```

Comment:
This does not seem to work since 'time' passes through the input variable 'tim'. The fact that `tim` is a linear function of `time` does not seem to be detected.
An option would be to use 'time' instead, but this would only avoid 76 state events for 1e8 seconds of simulation time.
55000 more events are generated by the `floor()` and `rem()` functions etc, but they do not seem to be able to generate time events at all. 
Even this model generates state events:
```
model test
  Real test;
equation
  test=floor(time);
end test;
```